### DISCUSSION TITLE
Why do scalar variables have components?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#23264] [Q&A General] [answered]
What are these used for? Are they different degrees of freedom? How is it different from a vector variable?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Scalar variables live in a single global location. There are two main differences:
- vector variables are currently fields, they have Qp spatially, while scalar variables have only a single value
- vector variables have 3 components (libmesh_dim really), while the scalar variables can have as many components as desired

Guillaume

**Reply by GiudGiud** [reply]
wrt to what they are used for, you ll have to imagine a problem with global variables that all have the same equation roughly

**Reply by jessecarterMOOSE** [reply]
So if I want to have multiple scalar variables, do I just define just one in my input file and use different components with different scalar kernels? Or can I define multiple scalar variables and use the 0'th component on each?

**Reply by GiudGiud** [reply]
it depends if the equations are very similar.
If you can write the equations in an "array" form, then it s really worth it to use scalar array variables.

If not then just use several scalar variables
